pkg/controller/daemon/daemoncontroller.go

```diff
@@ -510,6 +527,10 @@ func (dsc *DaemonSetsController) manage(ds *extensions.DaemonSet) error {
 	if err != nil {
 		return fmt.Errorf("couldn't get node to daemon pod mapping for daemon set %q: %v", ds.Name, err)
 	}
+	// Check DeletionTimestamp again because it may have been refreshed by getNodesToDaemonPods().
+	if ds.DeletionTimestamp != nil {
```

Comment:
We need to refresh the DS to check its latest deletionTimestamp _**after**_ listing all pods (i.e., potential controllees) at line 476. We do have a top-level check [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/daemon/daemoncontroller.go#L780), but that's before the refresh. The newly added checks are using the refreshed value.